src/pipecat/services/speechmatics/stt.py

```diff
@@ -0,0 +1,772 @@
+            )
+
+        # Complete configuration objects
+        self._transcription_config: TranscriptionConfig = None
+        self._process_config(transcription_config)
+
+        # STT client
+        self._client: Optional[AsyncClient] = None
+        self._audio_buffer: AudioBuffer = AudioBuffer(maxsize=10)
+        self._start_time: Optional[datetime.datetime] = None
+
+        # Current utterance speech data
+        self._speech_fragments: list[SpeechFragment] = []
+
+    async def start(self, frame: StartFrame):
+        """Called when the new session starts."""
+        await super().start(frame)
+        await self._connect()
+
+    async def stop(self, frame: EndFrame):
+        """Called when the session ends."""
+        await super().stop(frame)
+        await self._disconnect()
+
+    async def cancel(self, frame: CancelFrame):
+        """Called when the session is cancelled."""
+        await super().cancel(frame)
+        await self._disconnect()
+
+    async def run_stt(self, audio: bytes) -> AsyncGenerator[Frame, None]:
+        """Adds audio to the audio buffer and yields None."""
+        self._audio_buffer.write_audio(audio)
+        yield None
+
+    async def _run_client(self) -> None:
+        """Runs the Speechmatics client in a thread."""
+        await self._client.transcribe(
+            self._audio_buffer,
+            transcription_config=self._transcription_config,
+            audio_format=AudioFormat(
+                encoding=self._audio_encoding,
+                sample_rate=self.sample_rate,
+                chunk_size=self._chunk_size,
+            ),
+        )
+
+    async def _connect(self) -> None:
+        """Connect to the STT service."""
+        # Create new STT RT client
+        self._client = AsyncClient(
+            api_key=self._api_key,
+            url=_get_endpoint_url(self._base_url),
+        )
+
+        # Recognition started event
+        @self._client.on(ServerMessageType.RECOGNITION_STARTED)
+        def _evt_on_recognition_started(message: dict[str, Any]):
+            logger.debug(f"Recognition started (session: {message.get('id')})")
+            self._start_time = datetime.datetime.now(datetime.timezone.utc)
+
+        # Partial transcript event
+        @self._client.on(ServerMessageType.ADD_PARTIAL_TRANSCRIPT)
+        def _evt_on_partial_transcript(message: dict[str, Any]):
+            self._handle_transcript(message, is_final=False)
+
+        # Final transcript event
+        @self._client.on(ServerMessageType.ADD_TRANSCRIPT)
+        def _evt_on_final_transcript(message: dict[str, Any]):
+            self._handle_transcript(message, is_final=True)
+
+        # End of Utterance
+        @self._client.on(ServerMessageType.END_OF_UTTERANCE)
+        def _evt_on_end_of_utterance(message: dict[str, Any]):
+            logger.debug("End of utterance received from STT")
+            asyncio.create_task(self._send_frames(finalized=True))
```

Comment:
Instead of creating a new thread, I believe we should do like this:

```suggestion
            asyncio.run_coroutine_threadsafe(self._send_frames(finalized=True), self.get_event_loop())
```